types/src/v0/error.rs

```diff
@@ -0,0 +1,104 @@
+use core::fmt::Debug;
```

Comment:
I think having a single `error.rs` is likely to grow very large and become very confusing. Can we do `types/src/v0/error/state.rs` instead?

Reply:
I am now considering moving all these errors into their relevant modules inside impls. We can leave the global Error for now and remove it in a separate PR